RIOT-OS/RIOT
# shell: provide submodule for each shell command

### Contribution description


Rewrite sys/shell/commands/Makefile to provide a submodule for each shell command and use those modules in the command list. This improves user experience by allowing us to see which shell command modules are compiled in via `make info-modules` etc. and makes shell_commands less "special" and more like other modules with submodules, such as ztimer.

The introduced Makefile.dep should keep the existing behaviour of adding shell commands automatically based on which modules are enabled in the build.

### Testing procedure



Check that shell commands still exist and can be run from the shell in all examples and tests.


### Issues/PRs references
